packages/publisher/gcs/src/Config.ts

```diff
@@ -34,4 +35,9 @@ export interface PublisherGCSConfig {
    * Custom function to provide the key to upload a given file to
    */
   keyResolver?: (fileName: string, platform: string, arch: string) => string;
+  /**
+   * Optional Metadata for GCS Objects
+   * See https://cloud.google.com/storage/docs/metadata for more info.
+   */
+  metadata?: ConfigMetadata;
```

Comment:
This should probably be a function that takes in fileName + platform + arch and returns `metadata`. I've uploaded a lot of Electron apps over the years and you almost always want different caching rules for different artifacts.

i.e. `RELEASES` you shouldn't cache for that long at all, but `foo-1.2.3.exe` should be cached for years

Reply:
Seeing that there is no metadata option at all right now this PR would still be a significant improvement. 

If someone needs granular control they can create a PR iterated off of this.

Reply:
To avoid future breaking changes, or us maintainers being stuck with additional workload maintaining two options. This really should be done in the future proof way first.

The maintainer group is not that big and this may seem like a small thing to you but it all adds up over time.

Reply:
Fair 👍 I will enhance this feature.